File: lab-2.3/c#/DoctorStrange.cs
namespace CharacterLib
{
    public class DoctorStrange : IDescribable
    {
        private string name;
        private int magicLevel;

        public DoctorStrange()
        {
            name = "Unknown";
            magicLevel = 50;
        }

        public DoctorStrange(string name, int magicLevel)
        {
            this.name = name;
            this.magicLevel = magicLevel;
        }

        public DoctorStrange(DoctorStrange other)
        {
            name = other.name;
            magicLevel = other.magicLevel;
        }

        public string GetName()
        {
            return name;
        }

        public int GetPower()
        {
            return magicLevel;
        }

        public string Describe()
        {
            return $"Doctor Strange: {name}, Magic: {magicLevel}";
        }
    }
}

File: lab-2.3/c++/DoctorStrange.cpp
#include "DoctorStrange.h"

DoctorStrange::DoctorStrange() : name("Unknown"), magicLevel(50) {}

DoctorStrange::DoctorStrange(std::string n, int ml)
    : name(n), magicLevel(ml) {}

DoctorStrange::DoctorStrange(const DoctorStrange& other)
    : name(other.name), magicLevel(other.magicLevel) {}

std::string DoctorStrange::getName() const {
    return name;
}

int DoctorStrange::getPower() const {
    return magicLevel;
}

std::string DoctorStrange::describe() const {
    return "Doctor Strange: " + name + ", Magic: " + std::to_string(magicLevel);
}

IDescribable* DoctorStrange::clone() const {
    return new DoctorStrange(*this);
}